lld/test/ELF/aarch64-got-merging-icf.s

```diff
@@ -0,0 +1,94 @@
+// REQUIRES: aarch64
```

Comment:
As you've switched to `#`, switch to `# REQUIRES: aarch64` as well.
While here, can you add a file-level comment about the purpose of the test? Use `## `